wger/core/api/views.py

```diff
@@ -120,4 +127,29 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = WeightUnit.objects.all()
     serializer_class = WeightUnitSerializer
     ordering_fields = '__all__'
-    filter_fields = ('name', )
+    filter_fields = ('name',)
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    permission_classes = (DjangoObjectPermissions,)
+    throttle_classes = (UserRateThrottle,)
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
```

Comment:
The normal registration process also creates a default `userprofile` and gym. not sure if this is being done here.